sw/device/tests/sim_dv/pwrmgr_sleep_all_wake_ups_impl.c

```diff
@@ -7,61 +7,118 @@
 
 #include "sw/device/tests/sim_dv/pwrmgr_sleep_all_wake_ups_impl.h"
 
+#include "dt/dt_pwrmgr.h"
 #include "sw/device/lib/arch/device.h"
-#include "sw/device/lib/dif/dif_adc_ctrl.h"
 #include "sw/device/lib/dif/dif_pinmux.h"
 #include "sw/device/lib/dif/dif_pwrmgr.h"
 #include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/runtime/print.h"
+
+#include "sw/device/lib/testing/autogen/isr_testutils.h"
+
+#ifdef HAS_ADC_CTRL
+#include "dt/dt_adc_ctrl.h"
+#include "sw/device/lib/dif/dif_adc_ctrl.h"
+#endif
+#ifdef HAS_SENSOR_CTRL
+#include "dt/dt_sensor_ctrl.h"
 #include "sw/device/lib/dif/dif_sensor_ctrl.h"
+
+#include "sensor_ctrl_regs.h"
+#endif
+#ifdef HAS_SYSRST_CTRL
+#include "dt/dt_sysrst_ctrl.h"
 #include "sw/device/lib/dif/dif_sysrst_ctrl.h"
+#endif
+#ifdef HAS_USBDEV
+#include "dt/dt_usbdev.h"
 #include "sw/device/lib/dif/dif_usbdev.h"
-
-#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
-#include "pwrmgr_regs.h"
-#include "sw/device/lib/testing/autogen/isr_testutils.h"
+#endif
 
 static const uint32_t kPinmuxWkupDetector5 = 5;
 static const uint32_t kSensorCtrlEventIdx = 0;
 
-dif_adc_ctrl_t adc_ctrl;
-dif_aon_timer_t aon_timer;
-dif_pinmux_t pinmux;
+enum {
+  kPlicTarget = 0,
+};
+
+static const dt_pwrmgr_t kPwrmgrDt = 0;
+static_assert(kDtPwrmgrCount == 1, "this library expects exactly one pwrmgr");
+static const dt_pinmux_t kPinmuxDt = 0;
+static_assert(kDtPinmuxCount == 1, "this library expects exactly one pinmux");
+static const dt_rv_plic_t kRvPlicDt = 0;
+static_assert(kDtRvPlicCount == 1, "this library expects exactly one rv_plic");
+
 dif_pwrmgr_t pwrmgr;
 dif_rv_plic_t rv_plic;
-dif_sensor_ctrl_t sensor_ctrl;
-dif_sysrst_ctrl_t sysrst_ctrl;
-dif_usbdev_t usbdev;
 
+#define INIT_DIF_DT(__mod_name, __src, __dtname, __difname) \
+  dt_##__mod_name##_t __dtname =                            \
+      dt_##__mod_name##_from_instance_id(__src.inst_id);    \
+  dif_##__mod_name##_t __difname;                           \
+  CHECK_DIF_OK(dif_##__mod_name##_init_from_dt(__dtname, &__difname));
```

Comment:
Looks like the `__dtname` is not used at all, so perhaps removing it from macro param is better.

Reply:
It's not used outside the macro? If it's only used inside macro I think it should just be `CHECK_DIF_OK(dif_##__mod_name##_init_from_dt(dt_##__mod_name##_from_instance_id(__src.inst_id), &__difname));`